core/math/random.cpp

```diff
@@ -95,6 +95,107 @@ Variant Random::choice(const Variant &p_from) {
 	return Variant();
 }
 
+Variant Random::choices(const Variant &p_from, int count, const Array &p_weights, bool is_cumulative) {
+	int sum = 0;
+	Array cumulative_weights;
+	Array weights;
+	Array indices;
+	Array weighted_choices;
```

Comment:
If weights/indices can only be integers, I think you could use `PoolIntArray` instead (both here and as input). If you pass `Array`, Godot will do the conversion to `PoolIntArray` automatically. I think this will actually simplify validation down the road.

Reply:
How do I convert an `Array` to `PoolIntArray` ? I would need it for getting `dict.values()` .
It says `no suitable user-defined conversion from 'Array' to 'PoolIntArray' exists`

Reply:
Hmm yeah, looks like Godot doesn't provide such conversion. I'd just initialize in a loop.

```c++
Variant Random::choices(const Variant &p_from, int count, const PoolIntArray &p_weights, bool is_cumulative) {
	int sum = 0;
	Array cumulative_weights;
	PoolIntArray weights;
	Array indices;
	Array weighted_choices;

	if((p_from.get_type() == Variant::DICTIONARY) && p_weights.empty()){
		Dictionary dict = p_from;
		Array w = dict.values();
		for (int i = 0; i < w.size(); ++i) {
			weights.push_back(w[i]);
		}
	} else {
		weights = p_weights;
	}
```